index.js

```diff
@@ -16,7 +17,7 @@ class CleanTerminalPlugin {
      * or compiler events (compiler events are supported for backwards compatibility)
      */
     if (compiler.hooks) {
-      this.useCompilerHooks(compiler);
+      this.useCompilerHooks(compiler, this.beforeCompile);
     } else {
       this.useCompilerEvents(compiler);
     }
```

Comment:
Should we stop supporting the `events` API (I believe this only works for Webpack V3).